File: infra/src/main/java/mixter/infra/repositories/EventSessionRepository.java
package mixter.infra.repositories;

import mixter.domain.Event;
import mixter.domain.identity.Session;
import mixter.infra.EventStore;

import java.util.List;

public class EventSessionRepository extends AggregateRepository<Session> {

    public EventSessionRepository(EventStore eventStore) {
        super(eventStore);
    }

    @Override
    protected Session fromHistory(List<Event> history) {
        return new Session(history);
    }


}

File: infra/src/main/scala/mixter/infra/repositories/EventSessionRepository.scala
package mixter.infra.repositories

import mixter.domain.identity.Session
import mixter.infra.EventStore

class EventSessionRepository(override val store: EventStore) extends AggregateRepository[Session]{
  override protected def build(initialEvent: Session#InitialEvent, history: Seq[Session#AggregateEvent]): Session =
      Session(initialEvent, history)
}
